### PR TITLE
Add Java 23 specific FAT test

### PR DESCRIPTION
**Pull request by gjwatts**
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

### CODE REVIEW

**Comment by gjwatts** [comment]
#build

**Review by KyleAure — DISMISSED** [review]

File: dev/io.openliberty.java.internal_fat/src-reference/java23/io/openliberty/java/internal/package-info.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.java.internal;
```
Comment (KyleAure) [inline dev/io.openliberty.java.internal_fat/src-reference/java23/io/openliberty/java/internal/package-info.java:13]: In the misc repo you also have a `module-info.java` file.  Perhaps that should also be included here.

Comment (gjwatts) [reply dev/io.openliberty.java.internal_fat/src-reference/java23/io/openliberty/java/internal/package-info.java:13]: Good catch!  I have added that file here and for the others as well.

**Review by KyleAure — APPROVED** [review]

**Comment by gjwatts** [comment]
I ran my PB against the Java 23 platform build and that passed as well.  I need to wait until we get the Java 23 compiled WAR file used for this FAT out in DHE.

**Comment by gjwatts** [comment]
#libby

**Comment by jhanders34** [comment]
#libby

**Pull request merged**